src/SentryLogAdapter.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Sentry;
+
+use DateTimeInterface;
+use Psr\Log\LogLevel;
+use Sentry\Breadcrumb;
+use Sentry\Event;
+use Sentry\Severity;
+use Sentry\State\HubInterface;
+use Sentry\State\Scope;
+use Throwable;
+use Yiisoft\Yii\Sentry\Integration\Integration;
+
+final class SentryLogAdapter
+{
+    /**
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    protected ?string $environment = null;
+
+    /**
+     * @psalm-suppress PropertyNotSetInConstructor
+     *
+     * @description should represent the current version of the calling
+     *             software. Can be any string (git commit, version number)
+     */
+    protected ?string $release = null;
+
+    protected string $minLevel;
+
+    protected array $levels
+        = [
+            LogLevel::EMERGENCY => 0,
+            LogLevel::ALERT => 1,
+            LogLevel::CRITICAL => 2,
+            LogLevel::ERROR => 3,
+            LogLevel::WARNING => 4,
+            LogLevel::NOTICE => 5,
+            LogLevel::INFO => 6,
+            LogLevel::DEBUG => 7,
+        ];
+
+    public function __construct(
+        private HubInterface $hub,
+        YiiSentryConfig $config,
+    ) {
+        $this->minLevel = $config->getLogLevel() ?? LogLevel::ERROR;
+    }
+
+    /**
+     * @suppress PhanTypeMismatchArgument
+     */
+    public function log(string $level, string $message, array $context): void
+    {
+        /**
+ * @psalm-suppress MixedAssignment
+*/
+        $exception = $context['exception'] ?? $context['throwable'] ?? null;
+        unset($context['exception'], $context['throwable']);
+
+        $this->hub->withScope(
+            function (Scope $scope) use (
+                $exception,
+                $level,
+                $message,
+                $context,
+            ) {
+                if (!empty($context['extra'])) {
+                    /**
+            * @psalm-suppress MixedAssignment
+            */
+                    foreach ($context['extra'] as $key => $tag) {
+                        $scope->setExtra((string)$key, $tag);
+                    }
+                    unset($context['extra']);
+                }
+
+                if (!empty($context['tags'])) {
+                    /**
+            * @psalm-suppress MixedAssignment
+            */
+                    foreach ($context['tags'] as $key => $tag) {
+                        $scope->setTag((string)$key, (string)$tag);
+                    }
+                    unset($context['tags']);
+                }
+
+                if (is_array($context['fingerprint'] ?? null)) {
+                    $scope->setFingerprint(
+                        $this->formatFingerPrint((array)$context['fingerprint'])
```

Comment:
Isn't `$context['fingerprint']` already an array because of i`s_array` check above?